### ISSUE TITLE
mark=ball connects and fills the path

### ISSUE THREAD

**Issue by ynikitenko** [pgf-tikz/pgfplots#473]
I'm creating a simple graph. When I use

        mark=ball,
        ball color=black,

my path gets connected (the last and the first points) and filled (which i do not want). When I use the same TeX file with

        mark=*,
        mark options={
            fill=black,
        },

everything works as expected, and I get a simple graph. I believe markers should not influence the construction of the whole path.

I can provide my whole script if needed.

This behaviour happens to both of my installations of TeXLive, from 2021 and 2023. compat=1.18.

**Comment by muzimuzhi** [comment]
> I can provide my whole script if needed.

Yes please provide a [minimal working example](https://tex.meta.stackexchange.com/q/228).

**Comment by ynikitenko** [comment]
Well, it was trivial to create that. At least easier than to upload it here. Please change the extension to *tex*.

*addplot* or *addplot+* don't make a difference. Regular marks (the red plot) work as expected.

`\documentclass{standalone}
\usepackage{tikz}
\usepackage{pgfplots}
\pgfplotsset{compat=1.18}

\pgfplotsset{
    IBD/.style={
        color=black,
        mark=ball,
        ball color=black,
    },
    MC/.style={
        color=red,
        mark=*,
        mark options={
            fill=red,
        },
    },
}

\begin{document}
\begin{tikzpicture}
\begin{axis}[
    legend entries={MC,IBD}
]
\addplot [MC] {x^2};
\addplot [IBD] {x^3};
\end{axis}
\end{tikzpicture}

\end{document}`

[mwe.txt](https://github.com/pgf-tikz/pgfplots/files/14560516/mwe.txt)
![mwe](https://github.com/pgf-tikz/pgfplots/assets/1204978/e57df3c7-1522-498c-938b-8316b125c271)

**Comment by ynikitenko** [comment]
> > I can provide my whole script if needed.
> 
> Yes please provide a [minimal working example](https://tex.meta.stackexchange.com/q/228).

Thank you for giving me a link to what an MWE is; I think you could add it into a [template for new issues](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository).

**Comment by muzimuzhi** [comment]
The culprit is `ball color=black`, which is an `tikz` option. From its doc in the pgfmanual ([unofficial online version](https://tikz.dev/library-shadings#pgf./tikz/ball:color)), it seems what you get (a path is auto closed and then shaded) is expected behavior.

> `/tikz/ball color=⟨color⟩` (no default)
>
> This option sets the color used for the ball shading. It sets the `shade` and `shading=ball` options. Note that the ball will never “completely” have the color `⟨color⟩`. At its “highlight” spot a certain amount of white is mixed in, at the border a certain amount of black. Because of this, it also makes sense to say ball `color=white` or ball `color=black`.

In your use case `ball color=<color>` should be used inside `mark options={<options>}` to limit its scope to markers and to not influence the main ploting.

```tex
\documentclass[margin=5pt, varwidth]{standalone}
\usepackage{pgfplots}
\pgfplotsset{compat=1.18}

\begin{document}
\begin{tikzpicture}
  \begin{axis}
    % using inside "mark options"
    \addplot [mark=ball, mark options={ball color=red}] {x^2};
    % reproducible with just "ball color"
    \addplot [ball color=black] {x^3};
  \end{axis}
\end{tikzpicture}

% effect of "ball color" on simple un-closed tikz path
\begin{tikzpicture}
  \draw[ball color=red] (0,0) -- (1,1) -- (1,0);
\end{tikzpicture}
\end{document}
```

![image](https://github.com/pgf-tikz/pgfplots/assets/6376638/49ac7412-1bd5-4e6d-9969-47582d4d17ea)

**Comment by ynikitenko** [comment]
Thank you for the clarification.
I never thought that pgfplots can have such interference with TikZ.
I use the standard pgfplots manual from my TeXLive distribution, maybe you could add an example of a proper ball color there? 

Do you have access to the official manual, why does there have to be an unofficial one?

**Comment by muzimuzhi** [comment]
True this reflects some inconsistent design and maybe misleading naming, all inherited from pgf/tikz.

- `mark color` sets filling color of plot marks only, but `ball color` does three things. It sets shading, turns on shading mode, and sets shading color. The first two things are equivalent to `shading=ball`.
- Color of plot marks is only picked up when drawing plot marks, but shading mode immediately affects current path.
- Things may be a bit clearer if there's a separate `ball shading color` option. Then one can use
   ```tex
   \addplot [ mark color=..., ball shading color=..., mark=..., mark options={shading=ball} ]
   ```
- Other shading options like `top color` share similar issue with `ball color`. Their key names read like `xxx color`, but apart from color setting, they also select shading and turns on shading mode for current path.
